### PR TITLE
cppcheck performance fixes in filesystem/

### PR DESCRIPTION
**Pull request by tobbi**
This is mostly iterator pre/ postfix fixes, but also removing string initializations and moving things to initializer list.

### CODE REVIEW

File: xbmc/filesystem/MythSession.cpp [hunk 1]
```diff
@@ -358,14 +358,14 @@ void CMythSession::SetSeasonAndEpisode(const cmyth_proginfo_t &program, int *sea
   return;
 }
 
-CMythSession::CMythSession(const CURL& url) : CThread("MythSession")
+CMythSession::CMythSession(const CURL& url) : CThread("MythSession"):
+  m_hostname(url.GetHostName()),
+  m_username(url.GetUserName() == "" ? MYTH_DEFAULT_USERNAME : url.GetUserName()),
+  m_password(url.GetPassWord() == "" ? MYTH_DEFAULT_PASSWORD : url.GetPassWord())
```
Comment (ace20022) [inline xbmc/filesystem/MythSession.cpp]: Please if this is the order they are declared

**Comment by ace20022** [comment]
Rebase needed.

File: xbmc/filesystem/MythSession.cpp (hunk 1 above)
Comment (tobbi) [inline xbmc/filesystem/MythSession.cpp]: Sorry, what do you want me to do? Reorder? Leave it out?

Comment (ace20022) [inline xbmc/filesystem/MythSession.cpp]: Forgot a "check", sorry. Inits are done in the order the members are declared (in the header). If the order here is different the compiler prints a warning. I've checked it myself and it's ok ;)

**Comment by tobbi** [comment]
Done rebasing.

**Comment by ace20022** [comment]
jenkins build this please

**Comment by tobbi** [comment]
Jenkins build seems to be failing. And I think I know why. Gonna update the PR soon.

**Comment by tobbi** [comment]
jenkins build this please

**Comment by kib** [comment]
jenkins build this please

**Comment by tobbi** [comment]
Okay, the failed jenkins build were for the previous update. 

MythSession.cpp: In constructor 'XFILE::CMythSession::CMythSession(const CURL&)':
MythSession.cpp:361:69: error: expected '{' before ':' token
MythSession.cpp: At global scope:
MythSession.cpp:361:69: error: expected unqualified-id before ':' token
make[1]: *** [MythSession.o] Error 1
make: *** [xbmc/filesystem/filesystem.a] Error 2

I think I fixed that already.

File: xbmc/filesystem/TuxBoxDirectory.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ namespace XFILE
       virtual bool AllowAll() const { return true; }
       virtual DIR_CACHE_TYPE GetCacheType(const CURL& url) const { return DIR_CACHE_ALWAYS; };
     private:
-      bool GetRootAndChildString(const std::string strPath, std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString );
+      bool GetRootAndChildString(const std::string& strPath, std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString );
```
Comment (ace20022) [inline xbmc/filesystem/TuxBoxDirectory.h]: TuxBoxDirectory.cpp needs that change too.

Comment (tobbi) [inline xbmc/filesystem/TuxBoxDirectory.h]: Done!

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
and again:

jenkins build this please

**Pull request merged**